itests/qtest/src/test/java/org/apache/hadoop/hive/cli/TestTezTPCDS30TBPerfCliDriver.java

```diff
@@ -17,30 +17,32 @@
  */
 package org.apache.hadoop.hive.cli;
 
-import java.io.File;
-import java.util.Comparator;
-import java.util.List;
-
 import org.apache.hadoop.hive.cli.control.CliAdapter;
 import org.apache.hadoop.hive.cli.control.CliConfigs;
+import org.apache.hadoop.hive.cli.control.SplitSupport;
 import org.junit.ClassRule;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TestRule;
 import org.junit.runner.RunWith;
 import org.junit.runners.Parameterized;
 import org.junit.runners.Parameterized.Parameters;
+import org.junit.runners.model.Statement;
+
+import java.io.File;
+import java.util.Comparator;
+import java.util.List;
 
 @RunWith(Parameterized.class)
-public class TestTezPerfCliDriver {
+public class TestTezTPCDS30TBPerfCliDriver {
 
-  static CliAdapter adapter = new CliConfigs.TezPerfCliConfig(false).getCliAdapter();
+  static CliAdapter adapter = new CliConfigs.TezTPCDS30TBCliConfig().getCliAdapter();
 
   @Parameters(name = "{0}")
   public static List<Object[]> getParameters() throws Exception {
     List<Object[]> parameters = adapter.getParameters();
     parameters.sort(new C1());
-    return parameters;
+    return SplitSupport.process(parameters, TestTezTPCDS30TBPerfCliDriver.class, 10);
```

Comment:
If I remember well, it was around 20 minutes.

Reply:
20 minutes is also fine...we have quite a few [replication tests above 30 minutes right now](http://ci.hive.apache.org/job/hive-precommit/job/master/337/testReport/org.apache.hadoop.hive.ql.parse/)


note that in its current form it would not work because you need to have a separated "N_SPLITS" integer in the class source(like in other clidriver classes) for the splitter to work.

I think we can leave this out for now - 10 splits is a bit too many for a 20 minute test (2 or 3 would fit better)
in case of this test there is a bit more overhead per execution; because it will probably need to download and launch the metastore.

Reply:
Sounds good, I will remove it!

Reply:
Done https://github.com/apache/hive/pull/1347/commits/d50a2f6747fc89ba6a84a243c49b896437065fbc